azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/IArtifact.java

```diff
@@ -0,0 +1,51 @@
+/*
+  Copyright (c) Microsoft Corporation. All rights reserved.
+  Licensed under the MIT License. See License.txt in the project root for
+  license information.
+ */
+
+package com.microsoft.azure.toolkit.lib.common.model;
+
+import lombok.Getter;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+
+public interface IArtifact {
+    String getId();
+
+    @Nonnull
+    static FileArtifact fromFile(@Nonnull final File file) {
```

Comment:
IMO, library just need to define the Artifact interface and tooling will provide the imlpements as well as the from method

Reply:
Besides, move this method to `FileArtifact`? I think we should not use the implementation of interface in interface itself.